woocommerce/woocommerce-android
# Update code freeze tooling

This PR updates the code_freeze tooling:
 - Current version is now detected from the code and the new version calculated by the tool.
 - the main RELEASE-NOTES.txt is now updated during the code freeze.
 - the tags are added and pushed during the code freeze.

## To test
_Note_: This PR is related to wordpress-mobile/release-toolkit#48.
Since release-toolkit PR depends on the new version scheme for WPAndroid that is not in develop yet, the checks related to the branch are temporarily disabled. I'll make sure to update the release-toolkit PR to re-enable them before merging.

### Setup
1.1. Make sure you have the current beta branch in your local repository (currently `release/2.1`).
1.2. Run `bundle install`.

### Code freeze test
1.3. Run `bundle exec fastlane code_freeze`.
1.4. Verify that the proposed changes (new versions) are correct and confirm.
1.5. Verify that the new release branch has been created (currently it should be `release/2.2`).
1.6. Verify that the version name and code have been updated.
1.7. Verify that the codefrozen tag has been added to the milestone.
1.8 Verify that the branch protection has been set on GitHub.
1.9. Verify that the changelog file has been created in your home directory.
1.10. Verify that tags have been added and pushed.

### Clean up
1.11. Remove the branch protection for GitHub.
1.12. Delete the tags on GitHub and on your local repository.
1.13. Delete the branch on GitHub and on your local repository.
1.14. Remove the codefrozen tag from the milestone.

Update release notes:
